components/process/process.go

```diff
@@ -61,7 +63,7 @@ func (provider *Provider) start(ctx context.Context, procDir string, inputSpec s
 	}
 	program, err := whichQ.Run()
 	if err != nil {
-		return state{}, err
+		return state{}, errutil.WithHTTPStatus(http.StatusBadRequest, err)
```

Comment:
I agree that having internals know about http is not ideal. One pattern that I like is having a set of domain errors constructors that internally set an http status code, e.g. `NewValidationFailedErr()` would create an error object that had an `HTTPStatusCode: 400`.

Reply:
yup. right now, this is very much "do something that works and make sure all the badness has an easy-to-grep for string" :)